File: src/QuickFind.java
import java.util.Arrays;
import java.util.Scanner;

public class QuickFind{

    int[] data;

    QuickFind(int[] data){
        this.data = data;
    }

    boolean isConnected(int x, int y){
        if(data[x] == data[y]){
            return true;
        }
        return false;
    }

    void union(int p, int q) {

        int pVal = this.data[p];
        int qVal = this.data[q];

        for (int x = 0; x < data.length; x++) {
            if (data[x] == pVal) {
                data[x] = qVal;
            }

        }

    }

    public static void main(String[] args){

        System.out.print("Enter values separated by commas: ");
        Scanner sc = new Scanner(System.in);
        int[] input = Arrays.stream(sc.nextLine().split(",")).mapToInt(Integer::valueOf).toArray();

//      Test data: [0,1,2,3,4,5,6,7,8,9]
//
//      test operations = expected output
//
//      union(2,1) = [0,1,1,3,4,5,6,7,8,9]
//      isConnected(1,2) = True
//      isConnected(1,3) = False
//      union(1,3) = [0,3,3,3,4,5,6,7,8,9]
        QuickFind qf = new QuickFind(input);
        qf.union(2,1);

        Arrays.stream(qf.data).forEach(value -> System.out.print(value+" "));
        System.out.println();

        System.out.println("isConnected(1,2) = " + qf.isConnected(1,2));
        System.out.println("isConnected(1,2) = " + qf.isConnected(1,3));

        qf.union(1,3);

        Arrays.stream(qf.data).forEach(value -> System.out.print(value+" "));
        System.out.println();


    }


}

File: Python/QuickFind.py
class QuickFind:

    data = None

    def __init__(self,data=[]):
        self.data = data

    def isConnected(self,p,q):
        if(self.data[p] == self.data[q]):
            return True
        else:
            return False

    def union(self,p,q):

        pVal = self.data[p]
        qVal = self.data[q]

        for x in range(0,len(self.data)):
            # if x's id is equal to p's id
            if self.data[x] == pVal:
                # x's id is replaced with q's id
                self.data[x] = qVal


if __name__ == "__main__":

    # Test data: [0,1,2,3,4,5,6,7,8,9]

    # test operations = expected output
    
    #   union(2,1) = [0,1,1,3,4,5,6,7,8,9]
    #   isConnected(1,2) = True
    #   isConnected(1,3) = False
    #   union(1,3) = [0,3,3,3,4,5,6,7,8,9]

    inStr = input("Enter values separated by comma: ")
    inVal = list(map(lambda x: int(x),list(str(inStr).split(","))))
    qf = QuickFind(inVal)

    qf.union(2,1)
    print(qf.data)
    print(qf.isConnected(1,2))
    print(qf.isConnected(1,3))
    qf.union(1,3)
    print(qf.data)